src/native/managed/cdac/Microsoft.Diagnostics.DataContractReader.Contracts/Contracts/Loader_1.cs

```diff
@@ -196,6 +200,76 @@ bool ILoader.TryGetLoadedImageContents(ModuleHandle handle, out TargetPointer ba
         return true;
     }
 
+    private static uint AlignUp(uint value, uint alignment)
+    {
+        if (alignment == 0)
+            throw new ArgumentException("Alignment must be greater than zero.", nameof(alignment));
+        return (value + alignment - 1) & ~(alignment - 1);
+    }
+
+    private static bool IsMapped(Data.PEImageLayout peImageLayout)
+    {
+        return (peImageLayout.Flags & (uint)PEImageFlags.FLAG_MAPPED) != 0;
+    }
+
+    private TargetPointer FindNTHeaders(Data.PEImageLayout imageLayout)
+    {
+        Data.ImageDosHeader dosHeader = _target.ProcessedData.GetOrAdd<Data.ImageDosHeader>(imageLayout.Base);
+        return imageLayout.Base + (uint)dosHeader.Lfanew;
+    }
+
+    private TargetPointer RvaToSection(int rva, Data.PEImageLayout imageLayout)
+    {
+        TargetPointer ntHeadersPtr = FindNTHeaders(imageLayout);
+        Data.ImageNTHeaders ntHeaders = _target.ProcessedData.GetOrAdd<Data.ImageNTHeaders>(ntHeadersPtr);
+        Target.TypeInfo type = _target.GetTypeInfo(DataType.ImageNTHeaders);
+        int offset = type.Fields[nameof(Data.ImageNTHeaders.OptionalHeader)].Offset;
```

Comment:
The regular runtime data structures are in platform native format. These Image... data structures are always in little endian. Do we have any prior art in the cDAC for dealing with endianess?

This is not actual problem at the moment since all architectures supported by CoreCLR are little endian currently. cDAC should be designed to bridge endianess differences for future.